test/SlowTests/MailingList/ReduceAndTransform2.cs

```diff
@@ -1,144 +0,0 @@
-                                        Id = g.Select(a => a.Id).FirstOrDefault(a => a != null),
-                                        LastName = g.Select(a => a.LastName).FirstOrDefault(a => a != null),
-                                        Roles = g.SelectMany(a => a.Roles)
-                                    };
-            }
-        }
-
-        private class PersonnelTransformer
-            : AbstractTransformerCreationTask<PersonnelAll.Mapping>
-        {
-            public class Result
-            {
-                public string Id { get; set; }
-                public string FullName { get; set; }
-            }
-
-            public PersonnelTransformer()
-            {
-                TransformResults = results =>
-                    from result in results
-
-                    select new Result
-                    {
-                        Id = result.Id,
-                        FullName = result.LastName
-                    };
-
-            }
-        }
-
-        [Fact]
-        public void WillTransform()
-        {
-            using (var store = GetDocumentStore())
-            {
-                var persons = new[] { "Ayende", "Rahien", "Oren", "Enei", "Alias" };
-                var roles = new[] { "Administrator", "Programmer", "Support", "Guest", "Someone" };
-
-                foreach (var person in persons)
-                {
-                    using (var session = store.OpenSession())
-                    {
-                        var personnel = new Personnel() { LastName = person };
-                        session.Store(personnel);
-
-                        foreach (var role in roles)
-                        {
-                            var roleToStore = new PersonnelRole() { PersonnelId = personnel.Id, RoleId = string.Format("Roles/{0}", role) };
-                            session.Store(roleToStore);
-                        }
-                        session.SaveChanges();
-                    }
-                }
-
-                new PersonnelAll().Execute(store);
-                new PersonnelTransformer().Execute(store);
-
-                using (var session = store.OpenSession())
-                {
-                    // -- Dirty wait for stale
-                    session.Query<PersonnelAll.Mapping, PersonnelAll>()
-                                         .Customize(customization => customization.WaitForNonStaleResults())
-                                         .ToArray();
-
-                    // Without transform (will give 5 results)
-                    var results1 = session.Query<PersonnelAll.Mapping, PersonnelAll>().ToArray();
-
-                    // With transform (will give 1 to 4 results, depending on the weather?)
-                    var results2 = session.Query<PersonnelAll.Mapping, PersonnelAll>()
-                                        .TransformWith<PersonnelTransformer, PersonnelTransformer.Result>()
-                                        .ToArray();
```

Comment:
Restore candidate: no idea how it could return wrong results due to transformer usage, however we'll be better off having test checking multi map reduce index + query defined transformation